workers/python/kitchen_sink.py

```diff
@@ -104,16 +111,55 @@ async def handle_action(self, action: Action) -> Optional[Payload]:
         elif action.HasField("continue_as_new"):
             workflow.continue_as_new(action.continue_as_new.arguments)
         elif action.HasField("timer"):
-            await asyncio.sleep(action.timer.milliseconds / 1000)
+            await handle_awaitable_choice(
+                asyncio.sleep(action.timer.milliseconds / 1000),
+                action.timer.awaitable_choice,
+            )
         elif action.HasField("exec_activity"):
-            await launch_activity(action.exec_activity)
+            await handle_awaitable_choice(
+                launch_activity(action.exec_activity),
+                action.exec_activity.awaitable_choice,
+            )
         elif action.HasField("exec_child_workflow"):
             child_action = action.exec_child_workflow
             child = child_action.workflow_type or "kitchenSink"
             args = [RawValue(i) for i in child_action.input]
-            await workflow.execute_child_workflow(
-                child, id=child_action.workflow_id, args=args
+            await handle_awaitable_choice(
+                workflow.start_child_workflow(
+                    child, id=child_action.workflow_id, args=args
+                ),
+                child_action.awaitable_choice,
             )
+        elif action.HasField("set_patch_marker"):
+            if action.set_patch_marker.deprecated:
+                workflow.deprecate_patch(action.set_patch_marker.patch_id)
+                was_patched = True
+            else:
+                was_patched = workflow.patched(action.set_patch_marker.patch_id)
+
+            if was_patched:
+                return await self.handle_action(action.set_patch_marker.inner_action)
+        elif action.HasField("set_workflow_state"):
+            self.workflow_state = action.set_workflow_state
+        elif action.HasField("await_workflow_state"):
+            await workflow.wait_condition(
+                lambda: self.workflow_state.kvs.get(action.await_workflow_state.key)
+                == action.await_workflow_state.value
+            )
+        elif action.HasField("upsert_memo"):
+            pass  # Python doesn't have memo upserting
+        elif action.HasField("upsert_search_attributes"):
+            updates: list[SearchAttributeUpdate[Any]] = []
+            # TODO: Use RawValue after https://github.com/temporalio/sdk-python/issues/438
+            #  and avoid checking key by name
+            for k, v in action.upsert_search_attributes.search_attributes.items():
```

Comment:
We need to include the search attribute type with each search attribute item in this action IMO. Nowadays, search attribute setting needs to have key, value, _and_ type every time to avoid ambiguity only the server can resolve.

Reply:
You mean add the type to the proto? Could do that. Testing all those isn't really that important here though, which is why I wanted to just pass raw payload through, since we're not asserting anything equals anything else really, just sometimes generating one of these commands.

Reply:
One of the primary reasons we did typed search attributes is because nowadays, we require more than just key+value to set a search attribute. We require key+type+value.

Reply:
Just going to keep using the string check since the attrs have to be predefined and doing this is essentially the same amount of work & the generator/dsl doesn't actually care about different types.